plugins/bf4/bf4.cpp

```diff
@@ -0,0 +1,242 @@
+            peekProc((BYTE *) squad_offset_2 + 0x234, squad_leader); // Team value: US (United States); RU (Russia); CH (China).
+
+    // This prevents the plugin from linking to the game in case something goes wrong during values retrieval from memory addresses.
+    if (! ok) {
+        return false;
+    }
+
+    if (state) { // If not in-game
+        context.clear(); // Clear context
+        identity.clear(); // Clear identity
+        // Set vectors values to 0.
+        for (int i=0;i<3;i++) {
+            avatar_pos[i] = avatar_front[i] = avatar_top[i] = camera_pos[i] =  camera_front[i] = camera_top[i] = 0.0f;
+        }
+
+        return true; // This tells Mumble to ignore all vectors.
+    }
+
+    serverid[sizeof(serverid)-1] = 0; // NUL terminate queried C strings. We do this to ensure the strings from the game are NUL terminated. They should be already, but we can't take any chances.
+    std::string ServerID(serverid);
+    ServerID = jsonStringEscape(ServerID); // Replace '"' character with ' ' (space) to prevent JSON break.
+    std::ostringstream ocontext;
+    ocontext << " {\"Server ID\": \"" << ServerID << "\"}"; // Set context with server ID
+    context = ocontext.str();
+
+    std::wostringstream oidentity;
+    oidentity << "{";
+    host[sizeof(host)-1] = 0; // NUL terminate queried C strings. We do this to ensure the strings from the game are NUL terminated. They should be already, but we can't take any chances.
+    std::string Host(host);
+    Host = jsonStringEscape(Host);
+    if (Host.find("bot") == std::string::npos) // Set host string as empty if "bot" is found in it.
+        if (!Host.empty()) {
+            oidentity << std::endl << "\"Host\": \"" << host << "\""; // If it's not empty, set Host (IP:Port) in identity.
+        }
+
+    team[sizeof(team)-1] = 0; // NUL terminate queried C strings. We do this to ensure the strings from the game are NUL terminated. They should be already, but we can't take any chances.
+    std::string Team(team);
+    Team = jsonStringEscape(Team);
+    if (!Team.empty()) {
+        oidentity << std::endl;
+        if (Team == "US")
+            oidentity << "\"Team\": \"United States\""; // If team value is US, set "United States" as team in identity.
+        else if (Team == "CH")
+            oidentity << "\"Team\": \"China\""; // If team value is CH, set "China" as team in identity.
+        else if (Team == "RU")
+            oidentity << "\"Team\": \"Russia\""; // If team value is RU, set "Russia" as team in identity.
+    }
+
+    if (squad == 0) {
+        oidentity << std::endl << "\"Squad\": \"No\""; // If squad value is 0, set squad state to "No" in identity.
+    } else {
+        // If squad value is in a value between 1 and 26, set squad name in identity using NATO Phonetic alphabet.
+        if (squad == 1)
+            oidentity << std::endl << "\"Squad\": \"Alpha\"";
```

Comment:
Needs comma at the end, like was done for "Host".

Same for all the following Squad values.